src/preset.ts

```diff
@@ -564,6 +563,28 @@ function isInheritable(key: keyof ConfigurePreset | keyof BuildPreset | keyof Te
     return key !== 'name' && key !== 'hidden' && key !== 'inherits' && key !== 'description' && key !== 'displayName';
 }
 
+export function inheritsFromUserPreset(preset: ConfigurePreset | BuildPreset | TestPreset | PackagePreset | WorkflowPreset,
+    presetType: 'configurePresets' | 'buildPresets' | 'testPresets' | 'packagePresets' | 'workflowPresets', folderPath: string): boolean {
+
+    const originalUserPresetsFile: PresetsFile = getOriginalUserPresetsFile(folderPath) || { version: 8 };
+
+    const presetInherits = (presets: Preset[] | undefined, inherits: string | string[]) => presets?.find(p =>
+        Array.isArray(inherits)
+            ? inherits.some(inherit => inherit === p.name)
+            : inherits === p.name
+    );
+
+    if ((presetType === 'buildPresets' || presetType === 'testPresets' || presetType === 'packagePresets') &&
```

Comment:
We could avoid some duplication by pulling `(preset as ConfigurePresetInheritPreset).configurePreset` out into a variable before the if statement